src/components/DistanceRequest.js

```diff
@@ -290,14 +228,69 @@ function DistanceRequest({iou, iouType, report, transaction, mapboxAccessToken,
                     />
                 )}
             </View>
-            <Button
-                success
-                style={[styles.w100, styles.mb4, styles.ph4, styles.flexShrink0]}
-                onPress={navigateToNextPage}
-                isDisabled={_.size(validatedWaypoints) < 2 || hasRouteError || isOffline}
-                text={translate('common.next')}
-            />
-        </ScrollView>
+        </>
+    );
+
+    const content = (
+        <>
+            <View style={styles.flex1}>
+                <DraggableList
+                    data={waypointsList}
+                    keyExtractor={(item) => item}
+                    shouldUsePortal
+                    onDragEnd={({data}) => {
+                        const newWaypoints = {};
+                        _.each(data, (waypoint, index) => {
+                            newWaypoints[`waypoint${index}`] = lodashGet(waypoints, waypoint, null);
+                        });
+                        Transaction.updateWaypoints(iou.transactionID, newWaypoints);
+                    }}
+                    scrollEventThrottle={variables.distanceScrollEventThrottle}
+                    ref={scrollViewRef}
+                    renderItem={({item, drag, getIndex, isActive}) => {
+                        const index = getIndex();
+                        let descriptionKey = 'distance.waypointDescription.';
+                        let waypointIcon;
+                        if (index === 0) {
+                            descriptionKey += 'start';
+                            waypointIcon = Expensicons.DotIndicatorUnfilled;
+                        } else if (index === lastWaypointIndex) {
+                            descriptionKey += 'finish';
+                            waypointIcon = Expensicons.Location;
+                        } else {
+                            descriptionKey += 'stop';
+                            waypointIcon = Expensicons.DotIndicator;
+                        }
+
+                        return (
+                            <MenuItemWithTopDescription
+                                description={translate(descriptionKey)}
+                                title={lodashGet(waypoints, [`waypoint${index}`, 'address'], '')}
+                                icon={Expensicons.DragHandles}
+                                iconFill={theme.icon}
+                                secondaryIcon={waypointIcon}
+                                secondaryIconFill={theme.icon}
+                                shouldShowRightIcon
+                                onPress={() => Navigation.navigate(ROUTES.getMoneyRequestWaypointRoute('request', index))}
+                                onSecondaryInteraction={drag}
+                                focused={isActive}
+                                key={item}
+                            />
+                        );
+                    }}
```

Comment:
Would it be more cleaner if you extract this code to a separate function and use it here.